### ISSUE TITLE
Neovim died without answering (Ubuntu 22.04 fresh install)

### ISSUE THREAD

**Issue by ramuyk** [glacambre/firenvim#1445]
- OS Version: Ubuntu 22.04
- Browser Version: Firefox 106.0.2 (64-bit)
- Browser Addon Version: 0.2.13
- Neovim Plugin Version: v0.8.0-1210-gd367ed9b2

### What I tried to do

Hi, I've been using firenvim for a while already. This week I've decided to install a fresh install of Ubuntu 22.04 and after installing neovim and firenvim I'm running into trouble making firenvim work in Firefox (on Chrome it works fine)... I suspected it could have something to do with Firefox being installed with snap on Ubuntu 22.04 by default, then I tried to [remove the snap package and install it from its original repository](https://www.omgubuntu.co.uk/2022/04/how-to-install-firefox-deb-apt-ubuntu-22-04), but the behavior is still the same... I always see the message `Neovim died without answering` on Firefox.

I've checked the guide https://github.com/glacambre/firenvim/blob/master/TROUBLESHOOTING.md... The command `call firenvim#install(0)` works fine and the files are generated on my computer...  But the command `echo 'abcde{}' | $XDG_DATA_HOME/firenvim/firenvim` returns the error `zsh: no such file or directory: /firenvim/firenvim`. I'm not sure how to preceed from here...

**Comment by glacambre** [comment]
Hi, thanks for the report and for going through the troubleshooting guide. `$XDG_DATA_HOME` probably isn't defined in your shell - could you try `echo 'abcde{}' | $HOME/.local/share/firenvim/firenvim` instead?

**Comment by ramuyk** [comment]
Thanks for the response... 

You're right, `$XDG_DATA_HOME` is not defined on my shell, now I get the result:

```{"messages": [], "version": "0.2.13", "settings": {"localSettings": {".*": {"priority": 0, "cmdline": "neovim", "selector": "textarea, div[role=\"textbox\"]", "takeover": "never"}}}}```

Is that how it should be working?

**Comment by glacambre** [comment]
Yes, this looks fine.

Could you edit ~/.local/share/firenvim/firenvim as suggested [here](https://github.com/glacambre/firenvim/blob/master/TROUBLESHOOTING.md#print-debugging-your-initvim), attempt to use Firenvim a couple of times and then let me know if `/tmp/stdout` and `/tmp/stderr` exist, and if they do what their content is?

**Comment by ramuyk** [comment]
I've tried writing the following line at the end of `~/.local/share/firenvim/firenvim`:

```
exec '/home/myuser/path/to/nvim' --headless --cmd "let g:firenvim_i=[]|let g:firenvim_o=[]|let g:Firenvim_oi={i,d,e->add(g:firenvim_i,d)}|let g:Firenvim_oo={t->add(g:firenvim_o,t)}|let g:firenvim_c=stdioopen({'on_stdin':{i,d,e->g:Firenvim_oi(i,d,e)},'on_print':{t->g:Firenvim_oo(t)}})" --cmd 'let g:started_by_firenvim = v:true' -c 'call firenvim#run()' 2>>/tmp/stderr | tee -a /tmp/stdout
```

The files `/tmp/stout` and `/tmp/stderr` exist, but `/tmp/sderr` is empty and `/tmp/stdout` has the content:

```
{"messages": [], "version": "0.2.13", "settings": {"localSettings": {".*": {"priority": 0, "cmdline": "neovim", "selector": "textarea, div[role=\"textbox\"]", "takeover": "never"}}}}
```

![2022-10-31_20-15](https://user-images.githubusercontent.com/44415096/199127209-227dc7b2-4e35-4740-bf6a-4ec2272f351a.png)

**Comment by glacambre** [comment]
>The files /tmp/stout and /tmp/stderr exist, but /tmp/sderr is empty and /tmp/stdout has the content:

Was this after running Firenvim from the command line? I would expect to see a port number in this JSON object...

**Comment by glacambre** [comment]
Oh, I just noticed that your firenvim neovim plugin isn't up to date. Newer versions have better error reporting, so let's start from scratch. Can you:
- remove any firenvim configuration from your init.vim/lua
- update the firenvim neovim plugin to the latest version (currently b8f8645)
- remove ~/.local/share/firenvim/firenvim
- run `firenvim#install()` again
- append `2>>/tmp/stderr | tee -a /tmp/stdout` at the end of the `exec` line in the firenvim script
- close your browser
- remove /tmp/stderr and /tmp/stdout
- start your browser
- navigate to txti.es and click on the textarea a couple of times
- upload the content of /tmp/stdout and /tmp/stderr here

Hopefully we'll see either a port number or a proper error message.

**Comment by ramuyk** [comment]
Hi, I was going to follow the procedure you described now but I decided to completely remove Firefox from my system before it. After instaling it again firenvim was working. I really think it was something about `snap` that made those problems... When uninstalling it I used the commands `sudo apt-get remove --purge firefox` and `rm -rf ~/.mozilla`. After opening it with a fresh install and logging and synchronizing with my Firefox account firenvim was already working...

Thanks for the responses!
